cli/explain/compactgraph/compactgraph_test.go

```diff
@@ -16,141 +16,255 @@ import (
 )
 
 func TestJavaNoopImplChange(t *testing.T) {
-	spawnDiffs := diffLogs(t, "java_noop_impl_change", "7.3.1")
-	require.Len(t, spawnDiffs, 3)
-
-	sd1 := spawnDiffs[0]
-	assert.Regexp(t, "^bazel-out/[^/]+/bin/src/main/java/com/example/lib/liblib-hjar.jar$", sd1.PrimaryOutput)
-	assert.Equal(t, "//src/main/java/com/example/lib:lib", sd1.TargetLabel)
-	assert.Equal(t, "Turbine", sd1.Mnemonic)
-	assert.Empty(t, sd1.GetModified().GetTransitivelyInvalidated())
-	require.Len(t, sd1.GetModified().GetDiffs(), 1)
-	sd1d1 := sd1.GetModified().Diffs[0]
-	require.IsType(t, &spawn_diff.Diff_InputContents{}, sd1d1.Diff)
-	require.Len(t, sd1d1.GetInputContents().GetFileDiffs(), 1)
-	sd1fd1 := sd1d1.GetInputContents().GetFileDiffs()[0]
-	assert.Equal(t, "src/main/java/com/example/lib/Lib.java", sd1fd1.GetOldFile().GetPath())
-	assert.NotNil(t, sd1fd1.GetOldFile().GetDigest())
-	assert.NotEqual(t, sd1fd1.GetOldFile().GetDigest().GetHash(), sd1fd1.GetNewFile().GetDigest().GetHash())
-
-	sd2 := spawnDiffs[1]
-	assert.Regexp(t, "^bazel-out/[^/]+/bin/src/main/java/com/example/lib/liblib.jar$", sd2.PrimaryOutput)
-	assert.Equal(t, "//src/main/java/com/example/lib:lib", sd2.TargetLabel)
-	assert.Equal(t, "Javac", sd2.Mnemonic)
-	assert.Empty(t, sd2.GetModified().GetTransitivelyInvalidated())
-	require.Len(t, sd2.GetModified().GetDiffs(), 1)
-	sd2d1 := sd2.GetModified().Diffs[0]
-	require.IsType(t, &spawn_diff.Diff_InputContents{}, sd2d1.Diff)
-	require.Len(t, sd2d1.GetInputContents().GetFileDiffs(), 1)
-	sd2fd1 := sd2d1.GetInputContents().GetFileDiffs()[0]
-	assert.Equal(t, "src/main/java/com/example/lib/Lib.java", sd2fd1.GetOldFile().GetPath())
-	assert.NotNil(t, sd2fd1.GetOldFile().GetDigest())
-	assert.NotEqual(t, sd2fd1.GetOldFile().GetDigest().GetHash(), sd2fd1.GetNewFile().GetDigest().GetHash())
-
-	sd3 := spawnDiffs[2]
-	assert.Regexp(t, "^bazel-out/darwin_arm64-fastbuild/testlogs/src/test/java/com/example/lib/lib_test/test.log$", sd3.PrimaryOutput)
-	assert.Equal(t, "//src/test/java/com/example/lib:lib_test", sd3.TargetLabel)
-	assert.Equal(t, "TestRunner", sd3.Mnemonic)
-	assert.Empty(t, sd3.GetModified().GetTransitivelyInvalidated())
-	assert.True(t, sd3.GetModified().GetExpected())
-	require.Len(t, sd3.GetModified().GetDiffs(), 1)
-	sd3d1 := sd3.GetModified().Diffs[0]
-	require.IsType(t, &spawn_diff.Diff_OutputContents{}, sd3d1.Diff)
-	require.Len(t, sd3d1.GetOutputContents().GetFileDiffs(), 1)
-	sd3fd1 := sd3d1.GetOutputContents().GetFileDiffs()[0]
-	assert.Regexp(t, "^bazel-out/[^/]+/testlogs/src/test/java/com/example/lib/lib_test/test.xml$", sd3fd1.GetOldFile().GetPath())
+	t.Run("7.3.1", func(t *testing.T) {
```

Comment:
Nit: If the tests are different between versions, then it would be nicer to split them up into separate Test funcs. That way rules_go sharding logic could be applied neatly in the future.

If the tests are identical between versions, we can use a test table with nested t.Run() calls inside to mark with test is which.
```go
for _, bazelVersion := []string{"7.3.1", "8.0.0"} {
  t.Run(bazelVersion, func (t *testing.T) {
    for _, tc := testCases {
      t.Run(tc.Name, func (t *testing.T) { ... })
    }
  })
}
```

this way we can filter down the subtest with `--test_filter=`

Reply:
I split out the tests that are different depending on the version. The once that aren't already use `t.Run`.